internal/httphandlers/annotations_handler.go

```
@@ -73,12 +71,15 @@ func (d *eventsHandler) queryAnnotation(response http.ResponseWriter, request *h
 }
 
 func (d *eventsHandler) listAnnotations(response http.ResponseWriter, request *http.Request) {
-	params := mux.Vars(request)
-	typeId := params["typeId"]
-	ownerId := params["ownerId"]
-	linkId := params["linkId"]
+	typeId := mux.Vars(request)["typeId"]
+
+	// Get optional query parameters
+	query := request.URL.Query()
+	ownerId := query.Get("owner")
+	linkId := query.Get("link")
+	refId := query.Get("reference")
```

To preserve the earlier semantic you could return an error if owner & link & ref are all "".

---

or we can keep it open, that means a list of all offers... could be huge....
I think the range query on a key is a better mechanism to get all the records, because it supports paging. So I tend to not allowing it and require some filter there.

---

If they are all empty, then the query will return empty results.
This is because the query selector will be `{"$and": {}}` which returns nothing.